src/core/thread/link_quality.cpp

```diff
@@ -90,116 +72,107 @@ void LinkQualityInfo::AddRss(int8_t aNoiseFloor, int8_t aRss)
     // Multiply the the RSS value by a precision multiple (currently -8).
 
     newValue = static_cast<uint16_t>(-aRss);
-    newValue <<= kRssAveragePrecisionMultipleBitShift;
+    newValue <<= kPrecisionBitShift;
 
-    oldAverage = mRssAverage;
+    oldAverage = mAverage;
 
-    if (mCount >= kRssCountForWeightCoefficientOneEighth)
-    {
-        // New average = old average * 7/8 + new value * 1/8
-        mRssAverage = static_cast<uint16_t>(((oldAverage << 3) - oldAverage + newValue) >> 3);
-    }
-    else if (mCount >= kRssCountForWeightCoefficientOneFourth)
+    if (mCount == 0)
     {
-        // New average = old average * 3/4 + new value * 1/4
-        mRssAverage = static_cast<uint16_t>(((oldAverage << 2) - oldAverage + newValue) >> 2);
+        mCount++;
+        mAverage = newValue;
     }
-    else if (mCount >= kRssCountForWeightCoefficientOneHalf)
+    else if (mCount < kRssCountMax)
     {
-        // New average = old average * 1/2 + new value * 1/2
-        mRssAverage = (oldAverage + newValue) >> 1;
+        mCount++;
+
+        // New average = old average * (mCount -1)/mCount + new value * 1/mCount.
+        mAverage = static_cast<uint16_t>(((oldAverage * (mCount - 1)) + newValue) / mCount);
+
     }
     else
     {
-        mRssAverage = newValue;
-    }
-
-    if (mCount < kRssCountMax)
-    {
-        mCount++;
+        // New average = old average * 7/8 + new value * 1/8
+        mAverage = static_cast<uint16_t>(((oldAverage << 3) - oldAverage + newValue) >> 3);
```

Comment:
nit: magic number `3`.  Maybe `kRssCountMax` should be specified as a power of 2 to make things easier.